selfdrive/car/tests/test_lateral_limits.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+from collections import defaultdict
+import importlib
+from parameterized import parameterized_class
+import sys
+import unittest
+
+from common.realtime import DT_CTRL
+from selfdrive.car.car_helpers import interfaces
+from selfdrive.car.fingerprints import all_known_cars
+from selfdrive.car.interfaces import get_torque_params
+from selfdrive.car.hyundai.values import CAR as HYUNDAI
+
+CAR_MODELS = all_known_cars()
+
+# ISO 11270
+MAX_LAT_JERK = 2.5             # m/s^3
+MAX_LAT_JERK_TOLERANCE = 0.75  # m/s^3
+MAX_LAT_ACCEL = 3.0            # m/s^2
+
+# jerk is measured over half a second
+JERK_MEAS_TIME = 0.5  # seconds
+
+# TODO: update the max measured lateral accel for these cars
+ABOVE_LIMITS_CARS = [
+  HYUNDAI.KONA_EV,
+  HYUNDAI.KONA_HEV,
+  HYUNDAI.KONA,
+  HYUNDAI.KONA_EV_2022,
+]
+
+car_model_jerks = defaultdict(dict)
+
+
+@parameterized_class('car_model', [(c,) for c in CAR_MODELS])
+class TestLateralLimits(unittest.TestCase):
+  car_model: str
+
+  @classmethod
+  def setUpClass(cls):
+    CarInterface, _, _ = interfaces[cls.car_model]
+    CP = CarInterface.get_params(cls.car_model)
+
+    if CP.dashcamOnly:
+      raise unittest.SkipTest("Platform is behind dashcamOnly")
+
+    # TODO: test all platforms
+    if CP.lateralTuning.which() != 'torque':
+      raise unittest.SkipTest
+
+    if CP.notCar:
+      raise unittest.SkipTest
+
+    if CP.carFingerprint in ABOVE_LIMITS_CARS:
+      raise unittest.SkipTest
+
+    CarControllerParams = importlib.import_module(f'selfdrive.car.{CP.carName}.values').CarControllerParams
+    cls.control_params = CarControllerParams(CP)
+    cls.torque_params = get_torque_params(cls.car_model)
+
+  @staticmethod
+  def calculate_0_5s_jerk(control_params, torque_params):
+    steer_step = control_params.STEER_STEP
+    steer_up_per_frame = (control_params.STEER_DELTA_UP / control_params.STEER_MAX) / steer_step
+    steer_down_per_frame = (control_params.STEER_DELTA_DOWN / control_params.STEER_MAX) / steer_step
+
+    steer_up_0_5_sec = min(steer_up_per_frame * (JERK_MEAS_TIME / DT_CTRL), 1.0)
+    steer_down_0_5_sec = min(steer_down_per_frame * (JERK_MEAS_TIME / DT_CTRL), 1.0)
+
+    max_lat_accel = torque_params['MAX_LAT_ACCEL_MEASURED']
+    return steer_up_0_5_sec * max_lat_accel, steer_down_0_5_sec * max_lat_accel
```

Comment:
@HaraldSchafer I'm not sure if this is right, this is essentially the max lateral accel error (both directions) you could achieve in 0.5 seconds. If we multiplied by 2 to get over a second, it would be the same as the previous calculation